File: code/classes/pattern-matching.kt
fun describe(obj: Any): String =
    when (obj) {
        1 -&gt; "One"
        "Hello" -&gt; "Greeting"
        is Long -&gt; "Long number"
        !is String -&gt; "Not a string"
        else -&gt; "Unknown"
    }

fun main() {
    println(describe(1))
    println(describe("Hello"))
    println(describe(2L))
}

File: code/classes/pattern-matching.swift
// Pattern matching in Swift
func describe(obj: Any) -&gt; String {
    switch obj {
    case 1:
        return "One"
    case "Hello":
        return "Greeting"
    case is Double:
        return "Double number"
    default:
        return "Unknown"
    }
}

print(describe(obj: 1))
print(describe(obj: "Hello"))
print(describe(obj: 3.14))